LucachuTW/CARDS-PokemonPocket-scrapper
# Enhancement of Pokémon attack data extraction and energy management

## 📝 Enhancement of Data Extraction for Pokémon Cards

### Main Changes

#### ✨ Attack Extraction
- Implemented full extraction of energy costs for each attack
- Improved symbol-to-energy type mapping
- Optimized parsing of multi-word attack names

#### ⚡ Energy Management
- Added `type_mapping` dictionary for converting symbols to energy types
- Implemented `map_attack_cost` function for handling multiple energy costs
- Support for composite energy symbols (e.g., "GC")

#### 🔄 JSON Structure
```json
{
    "attacks": [
        {
            "name": "Attack Name",
            "cost": ["Grass", "Colorless"],
            "damage": "40"
        }
    ]
}
```

### 🧪 Testing
1. Run the script on cards with multiple attacks
2. Verify correct extraction of energy costs
3. Check the resulting JSON format

### 📚 Documentation
- Updated documentation with new output formats
- Added instructions for energy management
- Improved descriptions of main functions

### 🐛 Bug Fixes
- Enhanced error handling for unrecognized symbols
- Standardized output format
- Optimized cleaning of extracted data

These changes significantly improve the quality and accuracy of the data extracted from Pokémon cards, with a particular focus on attacks and energy costs.
